File: C#/78. Subsets.cs
public class Solution {
    public IList<IList<int>> Subsets(int[] nums) {
        IList<IList<int>> res = new List<IList<int>>();
        IList<int> temp = new List<int>();
        helper(0,nums,temp,res);
        return res;
    }
    public void helper(int i,int[] nums,IList<int> temp,IList<IList<int>> res){
        if(i==nums.Length){
            res.Add(new List<int>(temp));
            return;
        }
        temp.Add(nums[i]);
        helper(i+1,nums,temp,res);
        temp.Remove(nums[i]);
        helper(i+1,nums,temp,res);
    }
}

File: Java/78. Subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> res = new ArrayList<>();
        List<Integer> temp = new ArrayList<>();
        helper(0,nums,temp,res);
        return res;
    }
    public void helper(int i,int[] nums,List<Integer> temp,List<List<Integer>> res){
        if(i==nums.length){
            res.add(new ArrayList<Integer>(temp));
            return;
        }
        temp.add(nums[i]);
        helper(i+1,nums,temp,res);
        temp.remove(temp.size()-1);
        helper(i+1,nums,temp,res);
    }
}
